cpp/Naive.h

```diff
@@ -245,16 +243,24 @@ namespace ahr {
                            ky_(KY / 2 + 1) * bPerpMax / std::sqrt(1.0 + kperpDum2 * de * de);
 
             Real dx = lx / Real(X), dy = ly / Real(Y);
-            Real CFLFlow = std::min({dx / vxMax, dy / vyMax, 2.0 / omegaKaw,
-                                     std::min(dx / bxMax, dy / byMax) / (rhoS / de) / std::sqrt(LAST)});
+
+            Real CFLFlow;
+            if (M > 2) {
+                CFLFlow =
+                    std::min({dx / vxMax, dy / vyMax, 2.0 / omegaKaw,
+                              std::min(dx / bxMax, dy / byMax) / (rhoS / de) / std::sqrt(LAST)});
+            } else {
+                CFLFlow =
+                    std::min({dx / vxMax, dy / vyMax, 2.0 / omegaKaw, dx / bxMax, dy / byMax});
```

Comment:
I think in the 2 moment case we want LAST to have value 2. This is equivalent to ngtot in Julia/viriato, so this includes NE and APAR as the first two moments. So that last term in the min() would be kept.

Reply:
Okay so basically this can be the same as the original (no special handling for the `M=2` case, except we divide by `sqrt(M)` instead of `sqrt(LAST)`?

Reply:
Okay so actually I took the calculation from [here](https://github.com/Ozymandias314/adaptive-hermite-refinement/blob/f5743396e6e15aa8a7c95890f8ad75719f517379/viriato_julia/REGK.jl#L201). It seems that that term is omitted.

I also spotted a difference [here](https://github.com/Ozymandias314/adaptive-hermite-refinement/blob/f5743396e6e15aa8a7c95890f8ad75719f517379/viriato_julia/REGK.jl#L609): `CFL_flow` is calculated in much more simple terms, But I also assume it doesn't matter, because the estimated timestep is always larger than the one we use.

Reply:
Looks good